components/theme/Button.tsx

```diff
@@ -24,7 +39,9 @@ export const Button: React.FC<ButtonProps> = ({
   onClick = noop,
   children,
   size,
-  outline
+  outline,
+  testId,
+  ...other
 }) => {
   const classes = ['btn']
```

Comment:
What happens if someone passes in className as a prop? Will that override the current css of the button component, or will it just add on additional css?

Reply:
When I try to pass in `className` manually: this is the type error:
`"message": "Type '{ children: string; className: string; type: \"light\"; color: \"black\"; onClick: () => void; \"data-testid\": string; }' is not assignable to type 'IntrinsicAttributes & ButtonProps & { children?: ReactNode; }'.\n  Property 'className' does not exist on type 'IntrinsicAttributes & ButtonProps & { children?: ReactNode; }'.",
`
I've encountered this in the past when adding the ability to edit comments. This made me think i needed to add all the html attributes manually to the type definition(related to your other comment).

Reply:
I see, did my suggestion to use `interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>` work to solve this issue?

I tried this out and got no errors:
```
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  potato?: string
}

function Button({title, id, children, potato, className, ...other} : ButtonProps) {
  const classes = "potatoClass" + className || ''
	return <button title={title} id={id} className={classes} {...other}>{children}{potato}</button>
}

export default function App() {
	return <Button title="fakeTitle" id="fakeId" type="submit" potato=" amazing" data-testid="testId">Potato</Button>;
}
```

Reply:
One thing that will throw an error is `type`, because for buttons it can only have 3 values. But in our Button component in code its 
```
type?: 'success' | 'danger' | 'primary' | 'light' | 'info' | 'mute'
```

Reply:
> Also for `onClick`, we can remove explicitly defining that in button props(line 14) since it is an html attribute as well.

Extending the `ButtonProps` interface worked, but I'm going to keep the onClick explicitly defined in the button props, because leaving it out breaks the tests and it would make the PR too big. removing explicitly defining it would result in multiple snapshots being updated as well, so if it's something that's necessary I think it would be better to address it in a different PR.

Reply:
> This means we gotta get rename the `type` prop name to something else, as you suggested in the other thread. Also for `onClick`, we can remove explicitly defining that in button props(line 14) since it is an html attribute as well.

I was able address all your comments except for the onClick. I chose to explicitly state it because changing it was breaking many tests and would result in many snapshots having to be updated. This PR is already  big due to changing the `type` prop to `btnProp`(thus having to change it every file that used the `Button` component and passed in a `type` prop. I think if we want to focus on not explicitly defining `onClick` it should be in a future PR so that it will be easier to review.